src/platform/nrfconnect/FactoryDataProvider.h

```diff
@@ -0,0 +1,117 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <credentials/DeviceAttestationCredsProvider.h>
+#include <platform/CommissionableDataProvider.h>
+#include <platform/DeviceInstanceInfoProvider.h>
+
+#include <drivers/flash.h>
+#include <pm_config.h>
+
+extern "C" {
+#include "FactoryDataParser.h"
+}
+
+namespace chip {
+namespace DeviceLayer {
+
+struct InternalFlashFactoryData
+{
+    CHIP_ERROR GetFactoryDataPartition(uint8_t ** data, size_t & dataSize)
```

Comment:
nit: What do you think about renaming these arguments to better document the code?

```suggestion
    CHIP_ERROR GetFactoryDataPartition(uint8_t ** partitionAddress, size_t & partitionSize)
```